models/repo_language_stats.go

```diff
@@ -34,12 +51,41 @@ func (stats LanguageStatList) loadAttributes() {
 	}
 }
 
+func (stats LanguageStatList) getLanguagePercentages() map[string]float32 {
+	langPerc := make(map[string]float32)
+	var otherPerc float32 = 100
+	var total int64
+	for _, stat := range stats {
```

Comment:
There is no reason to range every language in `stat`, you can just range every language in `specialLanguages` and perform `delete` (which is no-op if doesn't exist) on `stats` with that key.

Reply:
That also means you'll be able to move `specialLanguages` from map to slice.

Reply:
I still need to loop to calculate total size & percentages in second loop